### PR TITLE
Switch Rediscala to Redis official client Lettuce core

### PR DESCRIPTION
**Pull request by zhaodaye2022**
RedisClusterSpec test ok
RedisStandaloneSpec test ok
RedisSentinelSpec test Ignored
RedisMasterSlavesSpec test fail ,I don't have the testing environment
Exception encountered when invoking run on a nested suite - org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Ready to accept connections tcp.*\n'
java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Ready to accept connections tcp.*\n'
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:68)
	at org.rnorth.ducttape.timeouts.Timeouts.doWithTimeout(Timeouts.java:60)
	at org.testcontainers.containers.wait.strategy.WaitAllStrategy.waitUntilReady(WaitAllStrategy.java:54)
	at org.testcontainers.containers.ComposeDelegate.waitUntilServiceStarted(ComposeDelegate.java:229)
	at java.base/java.util.concurrent.ConcurrentHashMap.forEach(ConcurrentHashMap.java:1603)
	at org.testcontainers.containers.ComposeDelegate.waitUntilServiceStarted(ComposeDelegate.java:203)
	at org.testcontainers.containers.ComposeContainer.start(ComposeContainer.java:140)
	at com.dimafeng.testcontainers.DockerComposeContainer.start(DockerComposeContainer.scala:184)
	at com.dimafeng.testcontainers.ContainerDef.start(ContainerDef.scala:14)
	at com.dimafeng.testcontainers.ContainerDef.start$(ContainerDef.scala:12)
	at com.dimafeng.testcontainers.DockerComposeContainer$Def.start(DockerComposeContainer.scala:88)
	at play.api.cache.redis.test.ForAllTestContainer.beforeAll(ForAllTestContainer.scala:23)
	at play.api.cache.redis.test.ForAllTestContainer.beforeAll$(ForAllTestContainer.scala:22)
	at play.api.cache.redis.connector.RedisMasterSlavesSpec.beforeAll(RedisMasterSlavesSpec.scala:12)
	at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:212)
	at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
	at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
	at play.api.cache.redis.connector.RedisMasterSlavesSpec.run(RedisMasterSlavesSpec.scala:12)
	at org.scalatest.tools.SuiteRunner.run(SuiteRunner.scala:47)
	at org.scalatest.tools.Runner$.$anonfun$doRunRunRunDaDoRunRun$13(Runner.scala:1321)
	at org.scalatest.tools.Runner$.$anonfun$doRunRunRunDaDoRunRun$13$adapted(Runner.scala:1315)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at org.scalatest.tools.Runner$.doRunRunRunDaDoRunRun(Runner.scala:1315)
	at org.scalatest.tools.Runner$.$anonfun$runOptionallyWithPassFailReporter$24(Runner.scala:992)
	at org.scalatest.tools.Runner$.$anonfun$runOptionallyWithPassFailReporter$24$adapted(Runner.scala:970)
	at org.scalatest.tools.Runner$.withClassLoaderAndDispatchReporter(Runner.scala:1481)
	at org.scalatest.tools.Runner$.runOptionallyWithPassFailReporter(Runner.scala:970)
	at org.scalatest.tools.Runner$.run(Runner.scala:798)
	at org.scalatest.tools.Runner.run(Runner.scala)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.runScalaTest2or3(ScalaTestRunner.java:43)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.main(ScalaTestRunner.java:26)
Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Ready to accept connections tcp.*\n'
	at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)
	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
	at org.testcontainers.containers.wait.strategy.WaitAllStrategy.waitUntilNestedStrategiesAreReady(WaitAllStrategy.java:66)
	at org.testcontainers.containers.wait.strategy.WaitAllStrategy.lambda$waitUntilReady$0(WaitAllStrategy.java:58)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

### CODE REVIEW

File: src/test/scala/play/api/cache/redis/test/RedisClusterContainer.scala [hunk 1]
```diff
@@ -13,7 +13,7 @@ trait RedisClusterContainer extends RedisContainer { this: Suite =>
 
   protected def redisSlaves = 1
 
-  final protected def initialPort = 7000
+  final protected def initialPort = 9001
```
Comment (KarelCemus) [inline src/test/scala/play/api/cache/redis/test/RedisClusterContainer.scala:16]: why is this changed?

File: build.sbt [hunk 2]
```diff
@@ -19,16 +19,18 @@ playVersion := "3.0.1"
 
 libraryDependencies ++= Seq(
   // play framework cache API
-  "org.playframework"   %% "play-cache"                % playVersion.value % Provided,
+  "org.playframework" %% "play-cache" % playVersion.value % Provided,
   // redis connector
-  "io.github.rediscala" %% "rediscala"                 % "1.14.0-pekko",
+  "io.lettuce" % "lettuce-core" % "6.3.2.RELEASE",
+  //Subsequent versions can attempt to remove it
+  "io.github.rediscala" %% "rediscala" % "1.14.0-pekko",
```
Comment (KarelCemus) [inline build.sbt:26]: Could you elaborate more on why which use is left to remove? Which code wasn't migrated?

**Review by KarelCemus — COMMENTED** [review]
Hello, nice PR, thank you for your contribution. I have two questions, if you wouldn't mind answering these, I would take over and finish it.

File: build.sbt (hunk 2 above)
Comment (zhaodaye2022) [reply build.sbt:26]: I have other services using port 7000 locally, you can restore it

Comment (zhaodaye2022) [reply build.sbt:26]: There are some dependencies related to spec testing that I have not made any changes to at the moment. After the subsequent changes are completed, I suggest deleting Redisscala because he is already dead

**Comment by zhaodaye2022** [comment]
How is the progress of this PR? I have updated some configuration logic on this PR in the future. Do you need to merge it

**Comment by KarelCemus** [comment]
Hey, sorry for delay, I am sorry for delay, these two weeks were very difficult for me. I already took a look and started on it, it just needs a bit more of my time, I'll try to do it by end of this week, I hope.

**Comment by zhaodaye2022** [comment]
Thank you very much for your work

**Comment by KarelCemus** [comment]
Closing this in favor of #301
